rest/plugins/watchlist.py

```diff
@@ -0,0 +1,159 @@
+                watched['dateAdded']=datetime.utcnow()
+                # Compute start and end dates
+                # default
+                end_date = datetime.utcnow() + timedelta(hours=1)
+                if duration == '12hr':
+                    end_date = datetime.utcnow() + timedelta(hours=12)
+                elif duration == '1d':
+                    end_date = datetime.utcnow() + timedelta(days=1)
+                elif duration == '2d':
+                    end_date = datetime.utcnow() + timedelta(days=2)
+                elif duration == '3d':
+                    end_date = datetime.utcnow() + timedelta(days=3)
+                elif duration == '1w':
+                    end_date = datetime.utcnow() + timedelta(days=7)
+                elif duration == '30d':
+                    end_date = datetime.utcnow() + timedelta(days=30)
+                watched['dateExpiring']=end_date
+                watched['comment']=comment
+                watched['creator']=userID
+                watched['reference']=referenceID
+                ref=watchlist.insert(watched)
+                sys.stdout.write('{0} written to db.\n'.format(ref))
+                sys.stdout.write('%s added to the watchlist table.\n' % (watchcontent))
+
+            else:
+                sys.stderr.write('%s is already present in the watchlist table\n' % (str(watchcontent)))
+        except Exception as e:
+            sys.stderr.write('Error while watching %s: %s\n' % (watchcontent, e))
+
+    def onMessage(self, request, response):
+        '''
+        request: http://bottlepy.org/docs/dev/api.html#the-request-object
+        response: http://bottlepy.org/docs/dev/api.html#the-response-object
+        '''
+        response.headers['X-PLUGIN'] = self.description
+
+        watchcontent = None
+        comment = None
+        duration = None
+        referenceID = None
+        userid = None
+        watchitem = False
+
+        # loop through the fields of the form
+        # and fill in our values
+        try:
+            for i in request.json:
+                # were we checked?
+                if self.name in i.keys():
+                    watchitem = i.values()[0]
+                if 'watchcontent' in i.keys():
+                    watchcontent = i.values()[0]
+                if 'duration' in i.keys():
+                    duration = i.values()[0]
+                if 'comment' in i.keys():
+                    comment = i.values()[0]
+                if 'referenceid' in i.keys():
+                    referenceID = i.values()[0]
+                if 'userid' in i.keys():
+                    userid = i.values()[0]
+
+            if watchitem and watchcontent is not None:
+                if len(watchcontent) < 2:
+                    sys.stderr.write('{0} does not meet requirements. Not added. \n'.format(watchcontent))
+
+                else:
+                    watchlisted = False
+                    if watchlisted is False:
```

Comment:
Is watchlisted ever not False?

Reply:
This can definitely be improved, I removed the logic that determined if an item is watchlisted, that's handled earlier in this code. So we could essentially remove the check for is false and simply input the data.